src/EnergyPlus/WindowEquivalentLayer.cc

```diff
@@ -151,16 +146,15 @@ void InitEquivalentLayerWindowCalculations(EnergyPlusData &state)
     state.dataWindowEquivalentLayer->EQLDiffPropFlag = true;
     state.dataWindowEquivalentLayer->CFSDiffAbsTrans = 0.0;
 
-    for (ConstrNum = 1; ConstrNum <= state.dataHeatBal->TotConstructs; ++ConstrNum) {
+    for (int ConstrNum = 1; ConstrNum <= state.dataHeatBal->TotConstructs; ++ConstrNum) {
         if (!state.dataConstruction->Construct(ConstrNum).TypeIsWindow) continue;
         if (!state.dataConstruction->Construct(ConstrNum).WindowTypeEQL) continue; // skip if not equivalent layer window
 
         SetEquivalentLayerWindowProperties(state, ConstrNum);
 
     } //  end do for TotConstructs
 
-    for (SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-        if (!state.dataConstruction->Construct(state.dataSurface->Surface(SurfNum).Construction).TypeIsWindow) continue;
+    for (int SurfNum : state.dataSurface->AllHTWindowSurfaceList) {
```

Comment:
In develop, this loops over all surfaces, including shading surfaces where `.Construction=0` causing an array bounds error. The fix reduces the loop to just window surfaces.

Reply:
Nice that all those pre-set lists are in place so it's an easy change like this.